optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -38,17 +38,17 @@
     private final DefaultSolverManager<Solution_, ProblemId_> solverManager;
     private final ProblemId_ problemId;
     private final Solver<Solution_> solver;
-    private final Function<ProblemId_, Solution_> problemFinder;
-    private final Consumer<Solution_> finalBestSolutionConsumer;
+    private final Function<ProblemId_, ? extends Solution_> problemFinder;
```

Comment:
Can be `? super ProblemId_`.

Reply:
Really good catches, Jiri. Those methods should indeed support accepting a Function<Object, ...> as a problemFinder for a SolverManager<Long, ...>.